client/src/app/pages/main-layout/main-layout.component.ts

```diff
@@ -1,92 +1,63 @@
+import { NgClass } from '@angular/common';
+import { Component, computed, inject, OnInit, signal } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router, RouterOutlet } from '@angular/router';
 import { KeycloakService } from '@app/core/services/keycloak/keycloak.service';
-import { PermissionService } from '@app/core/services/permission.service';
-import { injectQuery } from '@tanstack/angular-query-experimental';
 import { IconsModule } from 'icons.module';
 import { AvatarModule } from 'primeng/avatar';
 import { ButtonModule } from 'primeng/button';
 import { CardModule } from 'primeng/card';
 import { DividerModule } from 'primeng/divider';
 import { ToastModule } from 'primeng/toast';
 import { TooltipModule } from 'primeng/tooltip';
-import { HeliosIconComponent } from '../../components/helios-icon/helios-icon.component';
-import { UserLockInfoComponent } from '@app/components/user-lock-info/user-lock-info.component';
+import { FooterComponent } from '@app/components/footer/footer.component';
+import { NavigationBarComponent } from '@app/components/navigation-bar/navigation-bar.component';
+import { filter } from 'rxjs';
 
 @Component({
   selector: 'app-main-layout',
   standalone: true,
-  imports: [
-    RouterOutlet,
-    RouterLink,
-    SlicePipe,
-    ToastModule,
-    RouterLinkActive,
-    IconsModule,
-    ButtonModule,
-    TooltipModule,
-    HeliosIconComponent,
-    DividerModule,
-    AvatarModule,
-    CardModule,
-    ProfileNavSectionComponent,
-    UserLockInfoComponent,
-  ],
+  imports: [RouterOutlet, ToastModule, IconsModule, ButtonModule, TooltipModule, DividerModule, AvatarModule, CardModule, NgClass, FooterComponent, NavigationBarComponent],
   templateUrl: './main-layout.component.html',
 })
-export class MainLayoutComponent {
+export class MainLayoutComponent implements OnInit {
   private keycloakService = inject(KeycloakService);
-  private permissionService = inject(PermissionService);
+  private route = inject(ActivatedRoute);
+  private router = inject(Router);
 
-  username = computed(() => (this.keycloakService.decodedToken()?.preferred_username || '') as string);
+  repositoryId = signal<number | undefined>(undefined);
 
-  repositoryId = input.required({ transform: numberAttribute });
+  ngOnInit(): void {
+    // Initialize on first load (Refresh)
+    this.updateRepositoryId();
 
-  repositoryQuery = injectQuery(() => ({
-    ...getRepositoryByIdOptions({ path: { id: this.repositoryId() } }),
-    enabled: () => !!this.repositoryId(),
-  }));
+    // Listen for route changes (After initial load)
+    this.router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe(() => {
+      this.updateRepositoryId();
+    });
+  }
 
-  logout() {
-    this.keycloakService.logout();
+  private updateRepositoryId(): void {
+    let child = this.route.firstChild;
+
+    while (child?.firstChild) {
+      child = child.firstChild;
+    }
+
+    if (child) {
```

Comment:
suggestion: instead of nesting the logic here you can also go for the following as it makes the code more readable
```
if (!child) this.repositoryId....
const idFromSnapshot = child.snapshot.paramMap.get('repositoryId');
if (!idFromSnapshot) {
  this.repository.....
}
this.repositoryID......
```